.ecp-gitlab-ci.yml

```diff
@@ -33,19 +35,66 @@ stages:
 .Build:
   stage: build
   script:
-    - CI_PROJECT_DIR=${PWD}
     - module load cmake
-    - module load gcc
     - for i in ${BACKENDS}; do CMAKE_OPTS+=( -DCabana_ENABLE_${i}=ON ); done
     - j="$(grep -c processor /proc/cpuinfo 2>/dev/null)" || j=0; ((j++))
     - mkdir build && cd build &&
-      cmake -DCMAKE_PREFIX_PATH=${CI_PROJECT_DIR}/kokkos.install -DCabana_ENABLE_TESTING=ON -DCabana_ENABLE_Serial=OFF -DCabana_ENABLE_EXAMPLES=OFF ${CMAKE_OPTS[@]} .. &&
-      make -k -j${j} VERBOSE=1 &&
-      make test CTEST_OUTPUT_ON_FAILURE=1
+      cmake -DCMAKE_PREFIX_PATH=${CI_PROJECT_DIR}/kokkos.install -DCabana_ENABLE_TESTING=ON -DCabana_ENABLE_Serial=OFF -DCabana_ENABLE_EXAMPLES=ON ${CMAKE_OPTS[@]} ${CMAKE_EXTRA_ARGS} .. &&
+      make -k -j${j} VERBOSE=1
+    - wget https://raw.githubusercontent.com/slashbeast/pax-utils/master/lddtree.sh
   tags:
     - nobatch
+  artifacts:
+    paths:
+      - build/
+
+.Test:
+  stage: test
+  script:
+    - cd build
+    - make test CTEST_OUTPUT_ON_FAILURE=1
+  tags:
+    - batch
+
+.BuildKokkos_Cuda:
+  extends: .BuildKokkos
+  variables:
+    KOKKOS_EXTRA_ARGS: "--arch=Volta70,Power9 --with-cuda-options=enable_lambda"
+  before_script:
+    - module load gcc/6.4.0
+    - module load cuda
+
+.Build_Cuda:
+  extends: .Build
+  variables:
+    CMAKE_EXTRA_ARGS: "-DCMAKE_CXX_COMPILER=${CI_PROJECT_DIR}/kokkos.install/bin/nvcc_wrapper"
+  before_script:
+    - module load gcc/6.4.0
+    - module load cuda
+
+.Test_Cuda:
+  extends: .Test
+  before_script:
+    - module load gcc/6.4.0
+    - module load cuda
+
+BuildKokkos Cuda:
```

Comment:
Is this extra hierarchy needed here? Why not just use `BuildKokkos_Cuda` directly?

Reply:
Good point, it is not needed, but I had mulitple Cuda builds with different options in mind.

Reply:
And I wanted to keep it similar to the other backends.

Reply:
So do you want to keep this then or should we remove the extra layer for now and add it back when we have a need for additional build arrangements?

Reply:
I would like to keep it.